testsuite/tests/contrib/lkt_semantic/decl_block_correctness/test.lkt

```diff
@@ -0,0 +1,47 @@
+# Theses tests ensure that an error is emitted for invalid declarations in type
+# declaration.
+
+trait MyTrait {
+   @invalid val i: Int = 1
+   @invalid class MyClass {}
+   @invalid generic[T] class MyGenClass {}
+   @invalid trait InnerTrait {}
+   @invalid field: Int = 1
+
+   fun MyFun(i: Int): Int = i
+   generic[T] fun MyGenFun(i: T): T = i
+}
+
+enum MyEnum {
+   case a, b, c
+
+   @invalid val i: Int = 1
+   @invalid generic[T] class MyGenClass {}
+}
+
+struct MyStrut {
+   @invalid val i: Int = 1
+   @invalid struct InnerStruct {}
+   @invalid generic[T] fun MyGenFun(i: T): T = i
+
+   fun MyFun(i: Int): Int = i
+   field: Int = 1
+}
+
+class MyClass {
+   @invalid val i: Int = 1
+   @invalid class InnerClass {}
+   @invalid generic[T] trait MyTrait {}
```

Comment:
Could you check that generic field declarations are also rejected? (generic functions are allowed, regular fields are allowed, so checking generic fields looks worthwhile)